src/motioncore/protocols/boolean_gmw/boolean_gmw_gate.h

```diff
@@ -193,4 +193,28 @@ class MuxGate final : public ThreeGate {
   std::vector<std::unique_ptr<XcOtSender>> ot_sender_;
 };
 
+class ConstantAsBooleanGmwInputGate final : public motion::InputGate {
```

Comment:
is it the same as `SharedIn` in `Party`?

Reply:
Do you really need a gate for it or can you just create and register `Wire`s directly? At least for GMW. Unless I'm missing something, I think that the name of the protocol is unnecessary in the class/function name (in general). Better use namespaces or templates as in other code.